_posts/2020-11-26-torch-brain-segmentation/torch_brain_segmentation.Rmd

```diff
@@ -0,0 +1,705 @@
+---
+title: "Brain image segmentation with torch"
+description: > 
+  The need to segment images arises in various sciences and their applications, many of whom are vital to human (and animal) life. In this introductory post, we train a U-Net to mark lesioned regions on MRI brain scans.
+
+author:
+  - name: Sigrid Keydana
+    affiliation: RStudio
+    affiliation_url: https://www.rstudio.com/
+slug: keydanatorchtabular
+date: 11-26-2020
+categories:
+  - Torch
+  - R
+  - Image Recognition & Image Processing
+bibliography: bibliography.bib
+output:
+  distill::distill_article:
+    self_contained: false
+    toc: true
+preview: images/scans.png
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE, eval = FALSE)
+```
+
+## When *what* is not enough
+
+True, sometimes it's vital to distinguish between different kinds of objects. Is that a car speeding towards me, in which case I'd better jump out of the way? Or is it a huge terrier (in which case I'd probably do the same)? Often in real life though, instead of coarse-grained *classification*, what is needed is fine-grained *segmentation*.
+
+Zooming in on images, we're not looking for a single label; instead, we want to classify every pixel according to some criterion:
+
+-   In medicine, we may want to distinguish between different cell types, or identify tumors.
+
+-   In various earth sciences, satellite data are used to segment terrestrial surfaces.
+
+-   To enable use of custom backgrounds, video-conferencing software has to be able to tell foreground from background.
+
+Image segmentation is a form of supervised learning: Some kind of ground truth is needed. Here, it comes in form of a *mask* -- an image, of spatial resolution identical to that of the input data, that designates the true class for every pixel. Accordingly, classification loss is calculated pixel-wise; losses are then summed up to yield an aggregate to be used in optimization.
+
+The "canonical" architecture for image segmentation, around since 2015, is *U-Net.*
+
+## U-Net
+
+Here is the prototypical U-Net, as depicted in the original Rönneberger et al. paper [@RonnebergerFB15].
+
+Of this architecture, numerous variants exist; you could use different layer sizes, activations, ways to achieve downsizing and upsizing, and more. However, there is one defining characteristic: The U-shape, stabilized by the "bridges" crossing over horizontally at all levels.
+
```

Comment:
You've got two atypical sentence constructs in this paragraph (semicolon use is shockingly rare in American English, and the colon followed by an uppercase is also pretty uncommon). I'd ax one of them.

Reply:
Will do!

Reply:
just noticing, the uppercase shouldn't be there anyway ...